### PR TITLE
WordLoader Tests

### PR DESCRIPTION
**Pull request by jmk-198xtsuga**
Following along with momentum from FileLoader pull request (#14), wanted to get the framework build for WordLoader testing.  I've gotten a base framework in for that, marked with //TODO's for the things I have not yet implemented.

I am not sure what the continuous integration or other features of your Github.io site are or with automated pull requests, so I left the framework tests as //TODO items and did not mandate that they all fail() while they await implementation.

For tidiness: I did merge changes after your recent PR's, the only real difference i have is cleaning up where there was an extra script for test: jest in the NPM package file.

In WordLoader, to be able to import the module in the same way I did with FileLoader, I needed to add the "export" keyword to the class.

For the WordLoader tests, I have a simple test in place for the rendering of the widget, using Enzyme and Jest.  I haven't been able to master the process of requesting children of the React elements and inspecting them in more depth.  Other than that, I mostly set up the framework for individual tests of the methods, and of the behavior of the buttons (these are definitely duplications, so feel free to decide which one to keep)

Anyway, I think that will wrap up my work tonight, happy Halloween and happy Hacktoberfest!

### CODE REVIEW

**Comment by jmk-198xtsuga** [comment]
This PR addresses Issue #9 but does not completely resolve it, further test implementation would be appropriate

**Comment by sesamechicken** [comment]
@jmk-198xtsuga Can you pull master into your branch, pls? Pls squash and/or fixup your commits so we don't have merge commits in the history. 

Also, the // todo's should be tested. Commented code shouldn't be brought into `master`.

Thanks for all the help!

**Comment by jmk-198xtsuga** [comment]
backing up the changes in my own repo and will comment again when I can get my commits squashed or fixed up; changes currently available at https://github.com/jmk-198xtsuga/spelling-buddy/tree/testWordLoader

**Comment by jmk-198xtsuga** [comment]
I'll apologize for the force-push (I know, poor style) but it was the easier alternative as opposed to trying to figure out how many ways to reset/revert my master and testWordLoader branches after a few misguided attempts at git commands on my own system.

That said, this should now have a singular commit on top of my main branch that will apply, and I did pull your master branch to mine before committing so that should streamline the pull request.  The suggested diff to your master looks agreeable, and I don't believe I have the swampy mess in the git log anymore.

The testWordLoader file now has only defined tests (no //TODO: comments or framed out tests); I will experiment with those in my own forked repo on the testWordLoader branch if you would like to follow the progress (probably won't be very fast, may not even get much at all done by the end of the year... trying to self-learn React testing as I go, but I do enjoy the new frontiers and think it would be useful).

Ah, and somewhere along the way I reset back to a commit before the attempts to come up with better test description phrasing, just in case you were following that.  The attempted changes (still on my test branch) aren't very noteworthy.

Again, this certainly isn't a comprehensive test so further work is invited.

**Comment by jmk-198xtsuga** [comment]
again apologies on the force-push... I was trying to update my fork's Readme to direct any possible contributors over to this primary repo but it stacked the .gitignore commit onto this PR.  Would not be ideal because I was adding Readme.md to .gitignore.

To that extent, I'm going to not tinker too much with my Github repo until this PR is resolved, I'll keep work to local repos and other branches.

File: src/components/wordLoader/wordLoader.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { shallow } from "enzyme";
+import { Icon, Button, TextArea } from "semantic-ui-react";
+
+import { WordLoader } from "./index";
```
Comment (sesamechicken) [inline src/components/wordLoader/wordLoader.test.js:5]: We can simply import  from the index as: `import { WordLoader } from "./";` Also, the root level `jest.setup.js` file makes shallow, mount, and others globals so we don't have to import them in our test files.

File: src/components/wordLoader/wordLoader.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+describe("renders an expected Form", () => {
+    /** The "shallow" container for Enzyme to use for our React DOM "parent" */
+    let container;
+
+    /** Render our WordLoader Component for inspection. */
+    beforeAll(() => { container = shallow(<WordLoader />); } );
```
Comment (sesamechicken) [inline src/components/wordLoader/wordLoader.test.js:12]: This `beforeAll` is a major slowdown. We certainly don't need a new instance of the component for each test, particularly if we're just asserting that it has certain elements as children.

**Review by sesamechicken — CHANGES_REQUESTED** [review]
We need to test functionality of these UI elements, and not just their existence. Great start, though!

**Comment by jmk-198xtsuga** [comment]
Hey, sorry to be AFK so long, work has me busy this week.  I've seen the two comments and hope to address them soon this weekend

**Comment by sesamechicken** [comment]
> Hey, sorry to be AFK so long, work has me busy this week. I've seen the two comments and hope to address them soon this weekend

Hey no worries! I did start adding the unit tests for the wordLoader, so, you may want to fetch the changes to see what's going on. Also, if you're new to testing React, I can definitely help!

**Comment by sesamechicken** [comment]
I think the `wordLoader` units now cover everything except for `mapStateToProps` and `mapDispatchToProps`, which we don't need to test (`redux` already tests their code).

**Comment by jmk-198xtsuga** [comment]
Awesome to hear that you were able to fill out the tests!  In that case I'll consider the change request moot.

However, I'll still share this with ya: [Jest Setup and Teardown](https://jestjs.io/docs/en/setup-teardown)

Looks like beforeAll is a run-once call that the test runner calls and completes before running any tests, could help minimize code replication.  (beforeEach would be a more expensive one, especially if "rendering" these react elements each pass).  Might be reading that wrong, but thought I'd share the logic behind selecting a beforeAll call.

**Pull request closed without merging**